### PR TITLE
Add ExternalDependency annotation

### PR DESCRIPTION
**Pull request by Thushara-Piyasekara**
## Purpose
$subject
With this PR, a new annotation is introduced to annotate ballerina type definitions which are accessed through external Java dependencies.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/42205

## Approach
This new annotation was added to jballerina.java lang library. It can be accessed by importing jballerina.java in any ballerina module.

## Samples
```ballerina
import ballerina/jballerina.java;

@java:ExternalDependency
function foo(int x) returns int {
    return x * 2;
}
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/jballerina.java/src/main/ballerina/annotations.bal [hunk 1]
```diff
@@ -126,3 +126,12 @@ public const annotation FieldData FieldSet on source external;
 # }
 # ```
 public const annotation ObjectData Binding on class;
+
+# Identifies a type, class, or function that is used in the Java native code. This annotation should be added by the
+# developer if the type, class, or function could be accessed via an `io.ballerina.runtime.api.creators` API such as
```
Comment (prakanth97) [inline langlib/jballerina.java/src/main/ballerina/annotations.bal:131]:
```suggestion
# developer if the type, class, or function could be accessed via runtime APIs such as `ValueCreator`
```
WDYT?

File: tests/jballerina-unit-test/src/test/resources/test-src/annotations/external_dependency_annotation_negative.bal [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+// Types not allowed as sources for ExternalDependency annotation
```
Comment (prakanth97) [inline tests/jballerina-unit-test/src/test/resources/test-src/annotations/external_dependency_annotation_negative.bal:19]: Do we need this comment? shall we remove? I did not get the comment as well.

File: langlib/jballerina.java/src/main/ballerina/annotations.bal (hunk 1 above)
Comment (Thushara-Piyasekara) [reply langlib/jballerina.java/src/main/ballerina/annotations.bal]: I think its better. Refactored in 6d3dd2d12312dc2613b3dc83b67433e633510f18.

File: tests/jballerina-unit-test/src/test/resources/test-src/annotations/external_dependency_annotation_negative.bal (hunk 2 above)
Comment (Thushara-Piyasekara) [reply tests/jballerina-unit-test/src/test/resources/test-src/annotations/external_dependency_annotation_negative.bal]: Fixed in 6d3dd2d12312dc2613b3dc83b67433e633510f18.

**Review by prakanth97 — APPROVED** [review]
LGTM!

**Review by hasithaa — APPROVED** [review]

**Pull request merged**